### PR TITLE
fix: remove redundant final_activation layer in model.py

### PR DESCRIPTION
**Pull request by JCasaraconn**
Remove redundant final_activation layer in the forward() method of AbstractUNet.

The layer is redundant because it is only applied during validation, however the loss function of segmentation models already applies a final activation layer. Therefore the final_activation layer is applied twice in these instances.

### CODE REVIEW

**Comment by JCasaraconn** [comment]
I'm still thinking whether or not this means the validation eval should apply the final_activation layer, similar to the train eval as described here: https://github.com/wolny/pytorch-3dunet/blob/59170234fa902f946c7fa055e00396a054b6e9ca/pytorch3dunet/unet3d/trainer.py#L213-L222

I'm curious, what are your thoughts?

**Comment by wolny** [comment]
Hi @JCasaraconn. Thanks for you PR. 
Removing the final activation which returns the probabilities is not the right way to do it, because it's necessary during the inference.
I do agree with you that the logic is a bit convoluted and in fact may result in applying the sigmoid/softmax twice during the validation. I'm fixing this issue here: https://github.com/wolny/pytorch-3dunet/pull/121/commits/ba33ebc06260e3915e48b23d92a89099915bcb77. Basically the forward pass now returns both probabilities and logits.
